aiidalab_widgets_base/computational_resources.py

```diff
@@ -41,36 +45,47 @@ class ComputationalResourcesWidget(ipw.VBox):
     computers.
     """
 
-    value = traitlets.Unicode(allow_none=True)
-    codes = traitlets.Dict(allow_none=True)
-    allow_hidden_codes = traitlets.Bool(False)
-    allow_disabled_computers = traitlets.Bool(False)
-    default_calc_job_plugin = traitlets.Unicode(allow_none=True)
-
-    def __init__(self, description="Select code:", path_to_root="../", **kwargs):
+    value = tl.Unicode(allow_none=True)
+    codes = tl.Dict(allow_none=True)
+    allow_hidden_codes = tl.Bool(False)
+    allow_disabled_computers = tl.Bool(False)
+
+    def __init__(
+        self,
+        description="Select code:",
+        quick_setup=True,
+        detailed_setup=True,
+        clear_after=None,
+        default_calc_job_plugin=None,
+        **kwargs,
+    ):
         """Dropdown for Codes for one input plugin.
 
         description (str): Description to display before the dropdown.
         """
+        clear_after = clear_after or 15
+        self.default_calc_job_plugin = default_calc_job_plugin
         self.output = ipw.HTML()
-        self.setup_message = StatusHTML(clear_after=30)
+        self.setup_message = StatusHTML(clear_after=clear_after)
         self.code_select_dropdown = ipw.Dropdown(
             description=description,
             disabled=True,
             value=None,
             style={"description_width": "initial"},
         )
-        traitlets.directional_link(
+
+        # Create bi-directional link between the code_select_dropdown and the codes trait.
+        tl.directional_link(
             (self, "codes"),
             (self.code_select_dropdown, "options"),
             transform=lambda x: [(key, x[key]) for key in x],
         )
-        traitlets.directional_link(
+        tl.directional_link(
             (self.code_select_dropdown, "options"),
             (self, "codes"),
             transform=lambda x: {c[0]: c[1] for c in x},
         )
```

Comment:
You mean `ipw.dlink`? I assume `ipw.dlink` is the alias of `tl.directional_link`? correct me if I am wrong.
The reason that has to be `dlink` rather than the bilateral `tl.link` (which used for value below) is here the `codes` is a dictionary while options of `code_select_dropdown` contains tuple.

Reply:
> You mean `ipw.dlink`?

I actually meant the bi-directional `tl.link`


> The reason that has to be `dlink` rather than the bilateral `tl.link` (which used for value below) is here the `codes` is a dictionary while options of `code_select_dropdown` contains tuple.

Maybe we can use either a dictionary or a tuple? At least the dropdown accepts both.

Reply:
I would be more cautious to it, I found in https://github.com/aiidalab/aiidalab-widgets-base/pull/509 @danielhollas changes a lot of places where the dictionary was used for options to use iterable type (list or tuple in most cases). 
From my understanding, the dict is also treated as the iterable class but I'm not sure it won't raise the warning addressed by https://github.com/aiidalab/aiidalab-widgets-base/pull/509.

Reply:
Yep, I'll try to get that PR finished today so we can merge it, and then you can merge it to this branch.